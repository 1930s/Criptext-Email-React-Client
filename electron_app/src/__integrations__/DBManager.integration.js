/* eslint-env node, jest */

const DBManager = require('../DBManager');

beforeAll(async () => {
  await DBManager.cleanDataBase();
  await DBManager.createTables();
});

describe('Test DBManager', () => {
  it('should add email to db', async () => {
    await DBManager.createEmail({
      email: {
        key: '1',
        threadId: '1',
        s3Key: '1',
        unread: true,
        secure: true,
        content: 'Hello there',
        preview: 'Hello there',
        subject: 'Greetings',
        delivered: 0,
        date: '2013-10-07 08:23:19.120',
        isMuted: false
      }
    });
    const threadId = '1';
    const emails = await DBManager.getEmailsByThreadId(threadId);
    expect(emails).toMatchSnapshot();
  });

  it('should update thread emails as read', async () => {
    await DBManager.createEmail({
      email: {
        key: '2',
        threadId: '2',
        s3Key: '2',
        unread: true,
        secure: true,
        content: 'Hello there 2',
        preview: 'Hello there 2',
        subject: 'Greetings 2',
        delivered: 0,
        date: '2013-10-08 08:23:19.120',
        isMuted: false
      }
    });
    const threadId = '2';
    const unread = false;

    await DBManager.updateEmailByThreadId({ threadId, unread });
    const emails = await DBManager.getEmailsByThreadId(threadId);
    const updatedEmail = emails[0];
    expect(updatedEmail.unread).toBe(0);
  });

  it('should delete email', async () => {
    await DBManager.createEmail({
      email: {
        key: '3',
        threadId: '3',
        s3Key: '3',
        unread: true,
        secure: true,
        content: 'Hello there 3',
        preview: 'Hello there 3',
        subject: 'Greetings 3',
        delivered: 0,
        date: '2013-10-09 08:23:19.120',
        isMuted: false
      }
    });
    const threadId = '3';
    const key = '3';
    await DBManager.deleteEmailByKey(key);
    const emails = await DBManager.getEmailsByThreadId(threadId);
    expect(emails).toEqual([]);
  });
});
