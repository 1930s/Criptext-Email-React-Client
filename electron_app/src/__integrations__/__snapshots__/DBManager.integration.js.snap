// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DBManager should add email to db 1`] = `
Array [
  Object {
    "bcc": null,
    "cc": null,
    "content": "Hello there",
    "date": "2013-10-07 08:23:19.120",
    "delivered": 0,
    "from": null,
    "id": 1,
    "isMuted": 0,
    "key": "1",
    "preview": "Hello there",
    "s3Key": "1",
    "secure": 1,
    "subject": "Greetings",
    "threadId": "1",
    "to": null,
    "unread": 1,
  },
]
`;
